src/txHashBuilder.c

```diff
@@ -677,14 +829,15 @@ void txHashBuilder_addWithdrawal(
 
 static void txHashBuilder_assertCanLeaveWithdrawals(tx_hash_builder_t* builder)
 {
+	_TRACE("state = %u, remainingWithdrawals = %u", builder->state, builder->remainingWithdrawals);
+
 	ASSERT(builder->state <= TX_HASH_BUILDER_IN_WITHDRAWALS);
 
-	if (builder->state == TX_HASH_BUILDER_IN_WITHDRAWALS) {
-		ASSERT(builder->remainingWithdrawals == 0);
-	} else {
+	if (builder->state < TX_HASH_BUILDER_IN_WITHDRAWALS) {
```

Comment:
doing a `<` comparison against an enum feels kind of wrong and can lead to pretty weird bugs if new enums not necessarily respecting the ordering are introduced later on - wouldn't `!=` have the same effect?

Reply:
I was wary of inequalities here, but Pershing introduced them, and they make sense for a linear ordering. The alternative is to use a (potentially long) list of preceding states in a switch statement, and I don't like that much, since it would have to be modified if a new state is introduced in the linear ordering (not much likely to happen, though). 

If we make <= comparison (just above), then I find < also viable. Using != gives less information (one has to read it together with <= to actually know what the if condition describes).

Since the time the inequalities were introduced, I have added "substates" (for pool registration certificates and outputs), so the ordering is not as linear as it was before. I guess I should rewrite it into the explicit switch statement to be on the safe side.